substrate/client/db/src/lib.rs

```diff
@@ -2494,9 +2516,19 @@ impl<Block: BlockT> sc_client_api::backend::Backend<Block> for Backend<Block> {
 					let root = hdr.state_root;
 					let db_state =
 						DbStateBuilder::<HashingFor<Block>>::new(self.storage.clone(), root)
-							.with_optional_cache(
-								self.shared_trie_cache.as_ref().map(|c| c.local_cache()),
-							)
+							.with_optional_cache(self.shared_trie_cache.as_ref().map(|c| {
+								if self.unlimited_local_cache &&
```

Comment:
What happenes if we check only self.unlimited_local_cache here?

Reply:
It would enable for all context not just block authoring and importing.